tests/test_pipeline_invariants_v2.py

```
@@ -0,0 +1,174 @@
+                raise
+
+        task = asyncio.create_task(old_generation())
+        await asyncio.wait_for(registered.wait(), timeout=1)
+        await request_scope.start_new_document_request(state, kind="pretrial", mode="pretrial_waiting")
+        await asyncio.wait_for(cancelled.wait(), timeout=1)
+        try:
+            await task
+        except asyncio.CancelledError:
+            pass
+        else:
+            raise AssertionError("previous heavy generation task was not cancelled")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)